### DISCUSSION TITLE
List of definition token.lemma, token.dep abbrev used in doc/token

### DISCUSSION THREAD

**Discussion by js-compilatrum** [explosion/spaCy#9806] [Language Support]
I can't find in doc list of used abbrev. Some are obvious like _VERB/ROOT_, _ADJ_, but other not like _nmod:arg_, _obl:arg_, _advmod:emph_ is not. I ask here about data available for token inside doc.

Where can I found in documentation meaning of used shortcuts? Eventually what convention is used to figure out in other source what all at the mean?

Thank you for your suggestions. Best regards

**Comment by DuyguA** [comment]
Hellos, those are dependency tags and each model has its own labeling scheme, you can observe the list from the model's documentation page. This is English tags:
![tags](https://user-images.githubusercontent.com/8277232/144747877-84ea4468-6150-4416-aa65-4a5f7e835b51.png)

If you wanna know more about what dependency tagging mean, I can suggest a section from Jurafsky's book: https://web.stanford.edu/~jurafsky/slp3/14.pdf

Also universal dependencies contains information per tag along with examples how tags are used: https://universaldependencies.org/en/dep/index.html

Word lemma is the base form of a word, when you strip suffixes. It's the base form a word, when you strip the notion of time, person, singularity/plurality form this word.  Some examples:

```
went -> go
singing -> sing
children -> child
cats -> cat
```

Here's more explanation: https://simple.wikipedia.org/wiki/Lemma_(linguistics) . Also my book includes a section about lemma and stem concepts, in case you come across it in future.

Cheers!

**Reply by js-compilatrum** [reply]
Thank you for your contribution and explenation. It is what I was looking for! 

So to summary up tagger and parser shortcuts are typical for english linguistics (I ask especially on english models) and it is common standard not related to spacy conventions itself? Am I correct? It is my impression after studing "Speech and Language Processing. Daniel Jurafsky & James H. Martin" and provided "English Dependency Relation". I am sorry for that trivial question, but I want here seperate here conventions from NLP and language science (linguistics) itself to avoid mistake on deeper research as english is my secondary language.

**Reply by DuyguA** [reply]
Yes. Every language has its own tag set according to  its syntax and morphology (sometimes several tagsets). If someone provides a pretrained model to community, they specify which tagset they used. 

It's a very very good book, my recommendation to all NLP practitioners. If you have time I suggest reading the book from the beginning :wave:
